src/expr.rs

```diff
@@ -190,13 +190,21 @@ pub fn format_expr(
             rewrite_chain(expr, context, shape)
         }
         ast::ExprKind::Mac(ref mac) => {
-            rewrite_macro(mac, None, context, shape, MacroPosition::Expression).or_else(|| {
-                wrap_str(
-                    context.snippet(expr.span).to_owned(),
-                    context.config.max_width(),
-                    shape,
-                )
-            })
+            let should_skip = context
```

Comment:
We probably should check this inside `rewrite_macro` to avoid duplication between here and `visit_mac`.